### PR TITLE
[21357] Remove `TopicAttributes` from public APIs and make it private in `xmlparser`

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR:

* Removes `TopicAttributes` from public APIs in DDS and RTPS making usage of the new `PublicationBuiltinTopicData` and `SubscriptionBuiltinTopicData`.
* Moves `TopicAttributes` to the private folder and namespace `eprosima::fastdds::xmlparser`.
* Includes new `get_*_buitin_topic_data()` methods in `DataReader` and `DataWriter`
* Refactors some internal APIs (EDP, BuiltinProtocols,...) so they match the naming chain in snake_case.
* Includes an improvement in the discover server test to make it more resilient.

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- _N/A_ Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- _NO_: Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- _NO_: Changes are API compatible. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- New feature has been documented/Current behavior is correctly described in the documentation. 
    - [X] Related Discovery Server PR: https://github.com/eProsima/Discovery-Server/pull/100
    - Related documentation PR: `NOT NEEDED`
    - Related Python PR: `NOT NEEDED`
    - Related Shapes Demo PR: `NOT NEEDED`
- _N/A_ Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- _N/A_: If this is a critical bug fix, backports to the critical-only supported branches have been requested.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by MiguelCompany — DISMISSED** [review]
LGTM with green CI

**Review by MiguelCompany — APPROVED** [review]
LGTM with green CI

**Comment by Mario-DL** [comment]
@MiguelCompany commits LGTM

File: versions.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ Forthcoming
     * Several methods that were meant for internal use have been removed from public API
     * All public methods now have `snake_case` names
     * All public attributes now have `snake_case` names
+  * RTPSParticipant:
+    * Some methods changed to `snake_case`: `register_reader`, `register_writer`, `update_reader`, `update_writer`.
+	* Register methods take a `TopicDescription` instead of `TopicAttributes`.
```
Comment (Mario-DL) [inline versions.md:21]: * Update methods no longer take `TopicAttributes`

File: versions.md [hunk 2]
```diff
@@ -16,6 +16,9 @@ Forthcoming
     * Several methods that were meant for internal use have been removed from public API
     * All public methods now have `snake_case` names
     * All public attributes now have `snake_case` names
+  * RTPSParticipant:
```
Comment (Mario-DL) [inline versions.md:19]: We may also mention the new `get_*_builtin_topic_data()` in the DataWriter/Reader API

**Comment by Mario-DL** [comment]
LGTM, ci full passed in [4c3d06e](https://github.com/eProsima/Fast-DDS/pull/5085/commits/4c3d06ed9b4a525f44523329336be8d658471954)

**Review by MiguelCompany — APPROVED** [review]
* Full green CI on commit 4c3d06ed9b4a525f44523329336be8d658471954.
* Later changes are not modifying the code.

Ready to merge!

**Pull request merged**